src/renderer/components/notifications/notifications.store.tsx

```diff
@@ -19,6 +23,39 @@ export interface INotification {
   message: IMessage;
   status?: NotificationStatus;
   timeout?: number; // auto-hiding timeout in milliseconds, 0 = no hide
+  onClose?(): void; // additonal logic on when the notification times out or is closed by the "x"
+}
+
+export interface MainNotification {
+  title: string;
+  body: string;
+  buttons?: ({
+    backchannel: IpcChannel;
+  } & ButtonProps)[];
+  status: NotificationStatus;
+  timeout?: number;
+  closeChannel?: IpcChannel;
+}
+
+function renderButtons(id: IpcChannel, buttons?: MainNotification["buttons"]): React.ReactNode {
```

Comment:
Probably this should be react-component, like: `function RenderButtons(props: {id: IpcChannel, buttons?: MainNotification["buttons"]})`